microservices/kubeApi/clients/ocp_routes.py

```diff
@@ -174,16 +180,35 @@ def prepare_apply_routes(ns, select_tag, hosts, root_path, data_plane, ns_templa
 
             route_template = ROUTES[templ_version]["ROUTE"]
 
-            # If host transformation is disabled, then select the appropriate
-            # SSL cert based on the suffix mapping
+            # Modified certificate selection logic
             ssl_ref = "tls"
-            if not settings.host_transformation['enabled']:
+            custom_cert_label= ''
+            custom_cert_found = False
+            
+            if is_host_custom_domain(host):
+                logger.debug("[%s] Route A %03d Searching for custom cert for %s" % (select_tag, index, host))
+                if certificates:
+                    # Look for a matching certificate by SNI
+                    for cert in certificates:
+                        if host in cert['snis']:
+                            ssl_key = format_pem_data(cert['key'])
+                            ssl_crt = format_pem_data(cert['cert'])
+                            cert_id = cert['id']
+                            custom_cert_label = f'    aps-certificate-id: "{cert_id}"'
+                            logger.debug("[%s] Route A %03d Found custom cert with SNI match for %s" % (select_tag, index, host))
+                            custom_cert_found = True
+                            break
+                if not custom_cert_found:
+                    raise Exception("Custom certificate not found for host %s" % host)
+                
+            if not custom_cert_found and not settings.host_transformation['enabled']:
+                # Fall back to existing cert mapping logic
                 for host_match, ssl_file_prefix in host_cert_mapping.items():
                     if host.endswith(host_match):
                         ssl_ref = ssl_file_prefix
                         logger.debug("[%s] Route A %03d matched ssl cert %s" % (select_tag, index, ssl_ref))
-            ssl_key = read_and_indent("/ssl/%s.key" % ssl_ref, 8)
-            ssl_crt = read_and_indent("/ssl/%s.crt" % ssl_ref, 8)
+                ssl_key = read_and_indent("/ssl/%s.key" % ssl_ref, 8)
```

Comment:
what if host_transformation is enabled?  Where does `ssl_key` and `ssl_crt` get set to when custom_cert_found is False?